src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -884,36 +884,37 @@ export class BundleDataClient {
             // tokens to the filler. We can't remove non-empty message deposit here in case there is a slow fill
             // request for the deposit, we'd want to see the fill took place.
             .filter((fill) => fill.blockNumber <= destinationChainBlockRange[1] && !isZeroValueDeposit(fill)),
-          async (_fill) => {
+          async (fill) => {
             fillCounter++;
-            const relayDataHash = this.getRelayHashFromEvent(_fill);
+            const relayDataHash = this.getRelayHashFromEvent(fill);
             if (v3RelayHashes[relayDataHash]) {
               if (!v3RelayHashes[relayDataHash].fill) {
                 assert(
                   isDefined(v3RelayHashes[relayDataHash].deposit),
                   "Deposit should exist in relay hash dictionary."
                 );
-                // `fill` will only possibly differ from `_fill` in the `relayer` field, which does not affect the
-                // relay hash, so it is safe to modify.
-                const fill = await verifyFillRepayment(
```

Comment:
If I'm understanding this correctly, this issue was right here, where we would return before setting `v3RelayHashes` (since we did set `v3RelayHashes` before returning on incorrect repayment in the historical deposit case)? Also, the flow would be somebody fills with invalid relay data, we do not refund them here, but in the slow fill block, we don't see the invalid fill, so we create a leaf and send it to destination?

Reply:
In other words, returning early after finding that `fillToRefund` was undefined wasn't necessarily the issue (this would still be valid)
```
if (!isDefined(fillToRefund)) {
  bundleInvalidFillsV3.push(fill);
  return;
}
```
The issue was doing that before modifying `v3RelayHashes`?

Reply:
Ah I see.